### ISSUE TITLE
Connect TransportClient to Cluster

### ISSUE THREAD

**Issue by wolfgang-werner** [a-goryachev/docker-swarm-elasticsearch#1]
Hi, 

thanks for this - works great, but I have one issue - I cannot connect from the outside to the cluster communication. I currently do not want to run my client as a service to get the connection to the overlay network.

The access to Port 9200 via Docker Port works, but I would like to be able to connect java clients via transport protocool, that are not running inside the swarm-overlay network. I tried to publish port 9300 (as usal), but the client can not connect - I guess because of the internal publishing host.

I tried to set up a client profile (for elasticsearch to bind to different networks for internal (default profile) and clients, but this did not work also. 

Any comment would be very appreciated!

**Comment by a-goryachev** [comment]
Hi,

unfortunately you can not connect via transport protocol from outside. Because each container binds to internal network interface in order to handle transport connections. Look at this: https://github.com/a-goryachev/docker-swarm-elasticsearch/blob/master/es-docker#L44

You could access transport protocol by changing it like this: `network.publish_host=0.0.0.0` and use `--publish 9300:9300` option of `docker service create` command then. But it probably will not work due to routing mesh - each connection will be routed by docker to different container every time.

As a solution you could create service with `--mode global` option (to launch 1 container per host) and use something like `--publish bind=host,source=9300,target=9300,protocol=tcp` to publish port 9300 on each node in cluster and access each node from outside,

So finally try this:

1. Clone repository
2. Change "${MY_IP}" to "0.0.0.0" in https://github.com/a-goryachev/docker-swarm-elasticsearch/blob/master/es-docker#L44
3. Build an image `docker build -t my-es:latest .`
4. Push new image to docker hub or build it on each node.
5. Create a service: 
```
docker service create --name elasticsearch --network=elastic_cluster \
  --mode global \
  --env SERVICE_NAME=elasticsearch \
  --publish 9200:9200 \
  --publish bind=host,source=9300,target=9300,protocol=tcp \
  my-es:latest
```
6. Try to connect each node to port 9300.

**Comment by wolfgang-werner** [comment]
Thanks a lot. I tried this approach but was not successful. 

Some points
- your container works file with elasticsearch 5.3.0
- setting the published port to 0.0.0.0 does not work, in my case elasticsearch is publishing to the address 10.0.0.2 (I guess this is some type of broadcast / virtual service address, but leads to an error when starting the second node - publish address has to be unique over the nodes in elasticsearch cluster)

As my swarm is quite small, I now do different things
- for swarm-native solutions I use your initial setup, that works great if only internal cluster transport is needed
- for mixed environments, where I need two or more instances of elasticsearch, I just start the nodes with 'docker run', binding the transport port to the docker host's address and performing all network communication via the docker host own (primary) network.

Thanks again!
